WIP: Flake8 & isort

apollo13 opened the pull request:
I copied most of the settings from what we have in Django

spaceone commented on circuits/web/parsers/multipart.py:
```diff
@@ -32,34 +32,18 @@
 
 '''
-
-__author__ = 'Marcel Hellkamp'
```
Hm, this is just a package from upstream.

apollo13 commented:
So, the testsuite passes and I think I've addressed all comments  by @spaceone -- any objections to merging that? (Integrating web.url as dependency is imo out of scope for this issue).

spaceone commented:
You ignored some of my comments about the .decode('unicode_escape') changes

apollo13 commented:
@spaceone is that not addressed by https://github.com/circuits/circuits/pull/200/commits/ae187e6df19459af0765c1e59990e84fa5ebad98 ?

spaceone commented:
No, circuits.web.http is affected

spaceone commented:
And circuits.web.parsers.http

apollo13 commented:
Mhm, not sure I follow -- I've changed `circuits.web.parser.http`, but I think I've kept the original behavior of `bytes_to_str` there. The tests seem also okay (aside from changing them to adhere to the behavior change in b()).

Please tell me what I am missing

apollo13 commented on circuits/core/pollers.py:
```diff
@@ -57,7 +59,7 @@ def __init__(self, channel=channel):
         if platform.system() == "Linux":
             return os.pipe()
-        server = create_socket(AF_INET, SOCK_STREAM)
+        server = socket(AF_INET, SOCK_STREAM)
```
Nope, no idea why the `socket` class was aliased to `create_socket` :D

apollo13 commented on tests/app/test_daemon.py:
```diff
@@ -1,21 +1,24 @@
+if pytest.PLATFORM == "win32":
+    pytest.skip("Unsupported Platform")
+
+
```
Fixed, that said there is plenty of "too much whitespace" still around, but I am not going to fix all of those manually for this PR

apollo13 commented on circuits/node/utils.py:
```diff
@@ -13,7 +13,7 @@
     data = json.loads(s)
 
-    name = bytes_to_str(data["name"].encode("utf-8"))
+    name = data["name"]
```
So the current code is: https://github.com/circuits/circuits/pull/200/files#diff-015e99aa1b430ebfaa9aa46322a22df0R16 -- I think that fixes it

Essentially the idea is:
 * On Py3 classnames are unicode/text which is what json returns, so there is nothing to do
 * On Py2 classnames need to be bytes so we encode via utf-8

apollo13 commented on circuits/web/http.py:
```diff
@@ -22,22 +23,9 @@
-try:
-    unicode
-except NameError:
-    unicode = str
```
`unicode` was not used anywhere in this file and I do not think it was added as "utility" for other modules.

apollo13 commented on circuits/web/parsers/http.py:
```diff
@@ -187,7 +177,9 @@ def execute(self, data, length):
-                    first_line = bytes_to_str(b("").join(self._buf))
+                    first_line = b("").join(self._buf)
+                    if PY3:
+                        first_line = str(first_line, 'unicode_escape')
```
Okay, this takes a little bit -- lets step back:

What did `bytes_to_str(b("").join(self._buf))` originally do:
On py2 `bytes_to_str` is a no-op: https://github.com/circuits/circuits/blob/5797ea3769d8197a0ca3a4c7db898a740ce11a89/circuits/six.py#L299-L300
hence leaving it out is okay for python 2.

On Py3 it did decode the bytes using `unicode_escape`:
https://github.com/circuits/circuits/blob/5797ea3769d8197a0ca3a4c7db898a740ce11a89/circuits/six.py#L281-L282
which my `if PY3` should do.

(`bytes_to_str` is no longer part [or rather was never] of six, hence I removed it when updating six to upstream)

Arguably the code is broken (I am not sure that unicode_escape is the way to go here), but it is the same brokenness as before (or same correctness for that matter)

spaceone replied:
Alright. Then it is wrong on python3.

apollo13 commented on circuits/web/parsers/http.py:
```diff
@@ -313,7 +305,7 @@ def _parse_headers(self, data):
 
         # Split lines on \r\n keeping the \r\n on each line
-        lines = [bytes_to_str(line) + "\r\n"
+        lines = [(str(line, 'unicode_escape') if PY3 else line) + "\r\n"
```
Same answer as above
